src/Sarif/HashUtilities.cs

```diff
@@ -18,23 +18,31 @@ public static void ComputeHashes(string fileName, out string md5, out string sha
                 {
                     using (var bufferedStream = new BufferedStream(stream, 1024 * 32))
                     {
-                        var md5Cng = new MD5Cng();
-                        byte[] checksum = md5Cng.ComputeHash(bufferedStream);
-                        md5 = BitConverter.ToString(checksum).Replace("-", String.Empty);
+                        using (var md5Cng = new MD5Cng())
+                        {
+                            byte[] checksum = md5Cng.ComputeHash(bufferedStream);
+                            md5 = BitConverter.ToString(checksum).Replace("-", String.Empty);
 
-                        stream.Seek(0, SeekOrigin.Begin);
-                        bufferedStream.Seek(0, SeekOrigin.Begin);
 
-                        var sha1Cng = new SHA1Cng();
-                        checksum = sha1Cng.ComputeHash(bufferedStream);
-                        sha1 = BitConverter.ToString(checksum).Replace("-", String.Empty);
+                            stream.Seek(0, SeekOrigin.Begin);
+                            bufferedStream.Seek(0, SeekOrigin.Begin);
 
-                        stream.Seek(0, SeekOrigin.Begin);
-                        bufferedStream.Seek(0, SeekOrigin.Begin);
+                            using (var sha1Cng = new SHA1Cng())
+                            {
+                                checksum = sha1Cng.ComputeHash(bufferedStream);
+                                sha1 = BitConverter.ToString(checksum).Replace("-", String.Empty);
 
-                        var sha256Cng = new SHA256Cng();
-                        checksum = sha256Cng.ComputeHash(bufferedStream);
-                        sha256 = BitConverter.ToString(checksum).Replace("-", String.Empty);                            
+
+                                stream.Seek(0, SeekOrigin.Begin);
+                                bufferedStream.Seek(0, SeekOrigin.Begin);
+
+                                using (var sha256Cng = new SHA256Cng())
+                                {
+                                    checksum = sha256Cng.ComputeHash(bufferedStream);
+                                    sha256 = BitConverter.ToString(checksum).Replace("-", String.Empty);
+                                }
+                            }
+                        }
```

Comment:
You don't need multiple levels of nesting, because you only need each "Cng" object long enough to compute the checksum:

```
byte[] checksum;

using (var md5Cng = new MD5Cng())
{
   checksum = md5Cng.ComputeHash(bufferedStream);
   md5 = BitConverter.ToString(checksum).Replace("-", String.Empty);
}

stream.Seek(0, SeekOrigin.Begin);
bufferedStream.Seek(0, SeekOrigin.Begin);

using (var sha1Cng = new SHA1Chng())
{
    checksum = sha1Cng.ComputeHash(bufferedStream);
    sha1 = BitConverter.ToString(checksum).Replace("-", String.Empty);
}

// etc.
```